watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -428,6 +496,35 @@ def add_costing(m):
     dye_sep.nanofiltration.costing = UnitModelCostingBlock(
         flowsheet_costing_block=m.fs.zo_costing
     )
+
+    # TODO: Use MultiUnitModelCostingBlock
+    if hasattr(m.fs, "dewater"):
+        # dewater.dewaterer.costing = MultiUnitModelCostingBlock(
+        #     flowsheet_costing_block=m.fs.ro_costing,
+        #     costing_blocks={
+        #         "centrifuge": {
+        #             "costing_method": cost_centrifuge,
+        #             "costing_method_arguments": {"cost_electricity_flow": False},
+        #         },
+        #         "filter_belt_press": {
+        #             "costing_method": cost_filter_belt_press,
+        #             "costing_method_arguments": {"cost_electricity_flow": False},
+        #         },
+        #         "filter_plate_press": {
+        #             "costing_method": cost_filter_plate_press,
+        #             "costing_method_arguments": {"cost_electricity_flow": False},
+        #         },
+        #     },
+        #     initial_costing_block="centrifuge",
+        # )
```

Comment:
This MultiUnitCostingBlock was running into the following error and I haven't had the time to dig into it further:
![image](https://github.com/watertap-org/watertap/assets/96305519/17f0b821-75da-46ae-b6ee-806dce37a811)

I'll likely revisit this after I've generated some results/figures.

Reply:
Why add `dewaterer` after `dewater`? can you try  `m.fs.dewater.costing = MultiUnitModelCostingBlock `instead and see if you get the same error?

Reply:
`dewater` is the name of the `Block()` and `dewaterer` is the name of the actual unit model. It's identical to how costing was handled for the pretreatment section:
`    if hasattr(m.fs, "pretreatment"):
        prtrt.wwtp.costing = UnitModelCostingBlock(
            flowsheet_costing_block=m.fs.zo_costing
        )
    else:
        pass`

Reply:
I see... `dewater` is just a `Block`. In any case, because we are compartmentalizing in this fashion (not a bad thing), it looks like cost functions that use `blk.flowsheet()` would break since `flowsheet` attribute is higher up on the parent block. @bknueven - what do you think?

Reply:
Use an IDAES `FlowsheetBlock` instead of a Pyomo `Block`.

Reply:
Replacing `dewater = m.fs.dewater = Block()` with `dewater = m.dewater = FlowsheetBlock(dynamic=False)` or `dewater = m.fs.dewater = FlowsheetBlock(dynamic=False` still results in the same failure.

Reply:
@MarcusHolly just to clarify- is it true that you can get cost results without this error when you do not use the MultiUnitModelCostingBlock, but when you try using MultiUnitModelCostingBlock to obtain cost results, you see this error?

Reply:
Correct - the regular UnitModelCostingBlock is able to generate results